splunk-ingestable-s3-bucket/main.tf

```diff
@@ -0,0 +1,203 @@
+// S3 Bucket with a Lifecycle Policy that will delete old items
+resource "aws_s3_bucket" "this" {
+  bucket = "${var.log_bucket_name}"
+  acl    = "private"
+
+  lifecycle_rule {
+    id      = "auto-delete-after-${var.expiration_days}-days"
+    prefix  = ""
+    enabled = true
+
+    expiration {
+      days = "${var.expiration_days}"
+    }
+  }
+
+  tags = "${var.tags}"
+}
+
+// Policy for the bucket with READ and LIST permissions given to the read access account, Other accounts
+// Belonging to the organizaion are only allowed to Put objects to this bucket .
+
+resource "aws_s3_bucket_policy" "this" {
+  bucket = "${aws_s3_bucket.this.id}"
+
+  policy = <<POLICY
+{
+   "Version": "2012-10-17",
+   "Statement": [
+      {
+         "Sid": "splunk_access",
+         "Effect": "Allow",
+         "Principal": {
+            "AWS" : "arn:aws:iam::${var.read_access_account}:root"
+         },
+         "Action": [
+            "s3:Get*",
+            "s3:List*"
+         ],
+         "Resource": [
+            "arn:aws:s3:::${var.log_bucket_name}"
+         ]
+      },
+      {
+         "Sid": "all_accounts_in_organization_can_put_object_access",
+         "Effect": "Allow",
+         "Principal": "*",
+         "Action": [
+            "s3:Put*"
+         ],
+         "Condition": {
```

Comment:
That would solve it if your lambda is the only one writing. Let’s say I write an object (since the account I’m using has write access). I can then set permissions that does not grant bucket owner control. If we have this condition my write/put will fail. Might be nitpicking though, we use it now since it was in the AWS documentation for how to do cross account cloudtrail :-)

Reply:
@mikael-lindstrom  @itsdalmo  Resolved this in the lambda, and added the Condition to the bucket policy. Please review/apporve/merge.

Reply:
Great 👍